packages/core/src/utils/ConvertApiUtil.ts

```diff
@@ -0,0 +1,330 @@
+  sourceTokenAddress: string
+  toTokenAddress: string
+  sourceTokenAmount: string
+  fromAddress: string
+  decimals: number
+}
+
+export type TransactionData = {
+  from: string
+  to: `0x${string}`
+  data: `0x${string}`
+  value: string
+  gas: bigint
+  gasPrice: string
+}
+
+export type GetConvertDataResponse = {
+  toAmount: string
+  tx: TransactionData
+}
+
+export type GetGasPricesResponse = {
+  baseFree: string
+  low: {
+    maxPriorityFeePerGas: string
+    maxFeePerGas: string
+  }
+  medium: {
+    maxPriorityFeePerGas: string
+    maxFeePerGas: string
+  }
+  high: {
+    maxPriorityFeePerGas: string
+    maxFeePerGas: string
+  }
+  instant: {
+    maxPriorityFeePerGas: string
+    maxFeePerGas: string
+  }
+}
+
+// -- Controller ---------------------------------------- //
+export const ConvertApiUtil = {
+  get1InchAPI() {
+    const api = new FetchUtil({ baseUrl: ONEINCH_API_BASE_URL })
+    const chainId = CoreHelperUtil.getEvmChainId(NetworkController.state.caipNetwork?.id)
+    const { address } = AccountController.state
+
+    const endpoints = get1InchEndpoints(chainId, address)
+
+    return {
+      api,
+      paths: {
+        approveTransaction: endpoints.approveTransaction,
+        approveAllowance: endpoints.approveAllowance,
+        gas: endpoints.gasPrice,
+        gasPrice: endpoints.gasPrice,
+        swap: endpoints.swap,
+        tokens: endpoints.tokens,
+        tokensCustom: endpoints.tokensCustom,
+        tokenPrices: endpoints.tokensPrices,
+        search: endpoints.search,
+        balance: endpoints.balance,
+        quote: endpoints.quote
+      }
+    }
+  },
+
+  async getGasPrice() {
+    const { api, paths } = this.get1InchAPI()
+
+    const gasPrices = await api.get<GetGasPricesResponse>({
+      path: paths.gasPrice,
+      headers: { 'content-type': 'application/json' }
+    })
+
+    return gasPrices
+  },
+
+  async checkConvertAllowance({
+    fromAddress,
+    sourceTokenAddress,
+    sourceTokenAmount,
+    sourceTokenDecimals
+  }: GetAllowanceParams) {
+    const { api, paths } = this.get1InchAPI()
+
+    const res = await api.get<{ allowance: string }>({
+      path: paths.approveAllowance,
+      params: { tokenAddress: sourceTokenAddress, walletAddress: fromAddress }
+    })
+
+    if (res?.allowance && sourceTokenAmount && sourceTokenDecimals) {
+      const parsedValue = ConnectionController.parseUnits(sourceTokenAmount, sourceTokenDecimals)
+      const hasAllowance = BigInt(res.allowance) >= parsedValue
+
+      return hasAllowance
+    }
+
+    return false
+  },
+
+  async getTokenList() {
+    const { api, paths } = this.get1InchAPI()
+
+    return await api.get<TokenList>({ path: paths.tokens })
+  },
+
+  async searchTokens(searchTerm: string) {
+    const { api, paths } = this.get1InchAPI()
+
+    return await api.get<TokenInfo[]>({
+      path: paths.search,
+      params: { query: searchTerm }
+    })
+  },
+
+  async getMyTokensWithBalance() {
```

Comment:
isn't this available in portfolio endpoint cc @geekbrother @svenvoskamp ?

Reply:
Yeah, we switched to the same balance endpoint now.